Add image + action recording

rk1a opened the pull request:
Adds action recording

## To do

This PR is a Work in Progress 

- [x] some keys (mouse buttons, sneak, esc) are only recorded once when pressed, but we want to record the isDown state at every step

## How to test

1. `bash hacking_testing/server.sh`
2. `bash hacking_testing/record_client.sh`
3. `python data_recorder.py`

rk1a committed:
Restructure protobuf objects

rk1a committed:
Add action recording

rk1a committed:
Add debug mode

rk1a committed:
Fix action recording for long presses

rk1a committed:
Merge branch 'develop' into action-recording

rk1a commented on src/client/client.cpp:
```diff
@@ -1891,6 +1891,18 @@ void Client::makeScreenshot()
 
+float Client::getReward() {
+	float reward = 0.0;
+	//ClientScripting *scr = getScript();
```
This is commented because it is throwing LUA errors. @neverix I guess this is because I don't have the reward mod installed that you used?

neverix replied:
Probably, maybe the issue is in the config

neverix committed:
Add rewards clientmod

rk1a committed:
Remove optional flag

rk1a committed:
Build in RUN_IN_PLACE mode
this is to load config, mods etc. from the minetest folder
instead of user's home folder

rk1a committed:
Uncomment reward fetching from clientmods

rk1a committed:
Merge branch 'reward-client-mod' into action-recording

rk1a committed:
Warn when no reward mod installed
